sklearn/utils/fixes.py

```diff
@@ -323,3 +323,16 @@ def _object_dtype_isnan(X):
     from collections import Iterable as _Iterable  # noqa
     from collections import Mapping as _Mapping  # noqa
     from collections import Sized as _Sized  # noqa
+
+
+# backport for missing return_counts parameter in numpy.unique for numpy
+# versions < 1.9
+
+if np_version < (1, 9):
+    def _uniques_counts(array):
+        unique, idx = np.unique(array, return_inverse=True)
+        counts = np.bincount(idx).astype(float)
```

Comment:
Does it really return float counts?? That's surprising!

Reply:
At first I did it because int division in python 2 does not convert to float.
But eventually I used division from __future__ which seems a better way to achieve that, and forgot to remove the `.astype(float)`